test/validation-error-handling.test.js

```diff
@@ -336,3 +336,40 @@ test('should return a defined output message parsing JOI error details', t => {
     t.strictEqual(res.payload, '{"statusCode":400,"error":"Bad Request","message":"body \\"name\\" is required"}')
   })
 })
+
+test('should fail call custom error formatter', t => {
```

Comment:
Can you elaborate what kind of test cases we need here?

Reply:
here some examples: https://github.com/fastify/fastify/blob/615fb9b0ef040df5bd872c1ba8e2595f2cb3a88b/test/schema-feature.test.js#L457

when the user run `register()` a scope is created. so we need to define the tests:

Parent and child:
- register() + set error formatter X: the routes in this context will use X function
   - register() + set error formatter Y: the routes here will use Y function

Only Parent:
- register() + set error formatter X: the routes in this context will use X function
   - register(): the routes here will use X function

Only child:
- register(): the routes in this context will use the default function
   - register(): + set error formatter X: the routes in this context will use X function

Siblings:
- register() + set error formatter X: the routes in this context will use X function
- register() + set error formatter Y: the routes here will use Y function

Reply:
This is considering there is an ability to set `errorFromatter` per route? because this PR doesn't support this capability, only registering to a Fastify instance and providing an instance wide error formatter.